frontend/src/metabase/query_builder/components/expressions/ExpressionWidget.tsx

```diff
@@ -124,11 +124,24 @@ export const ExpressionWidget = (props: ExpressionWidgetProps): JSX.Element => {
             value={name}
             placeholder={t`Something nice and descriptive`}
             fullWidth
-            onChange={event => setName(event.target.value)}
+            onChange={event => {
+              // The expression's name shouldn't contain spaces. This checks
+              // handles if the user pastes a name with spaces.
+              if (event.target.value.includes(" ")) {
+                setName(event.target.value.replace(/\s/g, ""));
+              } else {
+                setName(event.target.value);
+              }
+            }}
```

Comment:
I believe it should do the trick

```suggestion
            onChange={event => setName(event.target.value.replace(/\s/g, "")}
```

Reply:
You're right! The check is not necessary